litestar/openapi/plugins.py

```diff
@@ -423,6 +433,21 @@ def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
                     </html>
                 """.encode()
 
+    def render_config(self) -> str:
+        """Render Scalar configuration to JS object."""
+        if not self.config:
+            return ""
+
+        dict_config = {}
+        for param, value in asdict(self.config).items():
+            dict_config[_camelize(param, False)] = value
```

Comment:
If you make the config an `msgspec.Struct` you could just do `json_config = msgspec.json.encode(self.config)`

```python
class ScalarConfig(msgspec.Struct, rename="camel"):
 ...
```